### PR TITLE
Build Docker image for sysl

### LINKED ISSUE #532: Docker image for go that sysl modules depends on
Currently sysl modules uses os.Execute() to get dependencies. 
This means that it's not consistent across go versions; currently It seems to work on go 12 but not go 13. 

Making a docker container would mitigate this problem as well as give it a better change of running in wasm distributions (as wasm distributions don't support the os package).

### PR DESCRIPTION
**Pull request by ChloePlanet**
Fix #532.

Changes proposed in this pull request:
- [x] Add Dockerfile. 
- [x] Declare Docker image in GoReleaser
- [x] Publish Docker image to package registries including Docker Hub<del>, Bintray, GitHub Packages, etc.</del>
- [x] Pack go binary into sysl Docker image

@anz-bank/sysl-developers

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -23,12 +23,38 @@ jobs:
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SYSL_PLANTUML: http://www.plantuml.com/plantuml
+
       - name: Validate goreleaser config
         uses: goreleaser/goreleaser-action@master
         with:
           args: check
 
-      - name: Release via goreleaser
+      - name: Login to Docker Hub
```
Comment (cuminandpaprika) [inline .github/workflows/release.yml]: A little bit confused about the docker hub login here. Is it necessary in order for goreleaser to upload the new docker image to docker.io?

Comment (ChloePlanet) [reply .github/workflows/release.yml]: Yes. goreleaser need this access to push image.

File: .goreleaser.yml [hunk 2]
```diff
@@ -6,24 +6,31 @@ before:
...
+  goarch:
+    - amd64
+    - 386
+    - arm
+    - arm64
+  goarm:
+    - 7
+
+dockers:
+  - image_templates:
```
[the remark this answers is not in the document]
Comment (ChloePlanet) [reply .goreleaser.yml]: As I tested, goreleaser will use linux/amd64 by default.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,5 @@
+FROM golang:1.13-alpine
```
[the remark this answers is not in the document]
Comment (ChloePlanet) [reply Dockerfile]: > LGTM, gotta say, really like the way you preface your commits with feat. and fix.

:)


> Also, when I was testing with `goreleaser release --rm-dist --skip-publish` I needed to do a `go mod tidy`, and commit before the goreleaser command would work.

Yep that's the way goreleaser works.

> Would love to see some comments here explaining what this Dockerfile is used for.

Agreed. Added.

**Pull request merged**